examples/vm-scale-set/virtual_machine_scale_set/loadbalancer-probe/main.tf

```diff
@@ -0,0 +1,125 @@
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "${var.prefix}-resources"
+  location = var.location
+}
+
+resource "azurerm_virtual_network" "example" {
+  name                = "${var.prefix}-vnet"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_subnet" "example" {
+  name                 = "${var.prefix}-subnet"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_public_ip" "example" {
+  name                = "${var.prefix}-pip"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  allocation_method   = "Dynamic"
+  domain_name_label   = azurerm_resource_group.example.name
+}
+
+resource "azurerm_lb" "example" {
+  name                = "${var.prefix}-lb"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  frontend_ip_configuration {
+    name                 = "PublicIPAddress"
+    public_ip_address_id = azurerm_public_ip.example.id
+  }
+}
+
+resource "azurerm_lb_backend_address_pool" "example" {
+  resource_group_name = azurerm_resource_group.example.name
+  loadbalancer_id     = azurerm_lb.example.id
+  name                = "BackEndAddressPool"
+}
+
+resource "azurerm_lb_probe" "example" {
+  resource_group_name = azurerm_resource_group.example.name
+  loadbalancer_id     = azurerm_lb.example.id
+  name                = "ssh-probe"
+  protocol            = "Tcp"
+  port                = 22
+}
+
+resource "azurerm_lb_nat_pool" "example" {
+  resource_group_name            = azurerm_resource_group.example.name
+  name                           = "ssh"
+  loadbalancer_id                = azurerm_lb.example.id
+  protocol                       = "Tcp"
+  frontend_port_start            = 220
+  frontend_port_end              = 229
+  backend_port                   = 22
+  frontend_ip_configuration_name = "PublicIPAddress"
+}
+
+resource "azurerm_virtual_machine_scale_set" "example" {
```

Comment:
can we switch this over to using `azurerm_linux_virtual_machine_scale_set` - the older resources have been superseded and shouldn't be used in examples going forward

Reply:
updated. thx